### ISSUE TITLE
Inconsistency doc / code: predict_log_density

### ISSUE THREAD

**Issue by AthenaisGautier** [GPflow/GPflow#2121]
# Documentation/tutorial notebooks

When comparing [the documentation of "predict_log_density"](https://gpflow.github.io/GPflow/2.9.1/api/gpflow/likelihoods/index.html#gpflow.likelihoods.Likelihood.predict_log_density) to source code, I found the following statement:

     Given a Normal distribution for the latent function, and a datum Y, compute the log predictive density of Y,

        i.e. if
            q(F) = N(Fmu, Fvar)

        and this object represents

            p(y|F)

        then this method computes the predictive density

            log ∫ p(y=Y|F)q(F) df

        :param X: input tensor
        :param Fmu: mean function evaluation tensor
        :param Fvar: variance of function evaluation tensor
        :param Y: observation tensor
        :returns: log predictive density

while the [code for predict_log_density in the Gaussian class](https://gpflow.github.io/GPflow/2.9.1/_modules/gpflow/likelihoods/scalar_continuous.html#Gaussian) is: 

        def _predict_log_density(
          self, X: TensorType, Fmu: TensorType, Fvar: TensorType, Y: TensorType
        ) -> tf.Tensor:
          return tf.reduce_sum(logdensities.gaussian(Y, Fmu, Fvar + self._variance(X)), axis=-1)

which, as far as I understand, corresponds to ∫ log p(y=Y|F)q(F) df (i.e. expected log-density) ?

Is this a lack of consistency between the documentation and code, or am I just l misunderstanding ?
Thanks and cheers!

**Comment by uri-granta** [comment]
Sorry for the delayed response! Good spot: it certainly looks like we're returning the predictive log density rather than the log predictive density as claimed in the docstring. I will check with someone who understands this better than me and post an update...

**Comment by uri-granta** [comment]
Ok, my understanding is that the docs are correct but perhaps incomplete. The code is calculating the log predictive density for each output and summing them (equivalent to the the log of the product). The integral alluded to in the docs is actually inside `logdensities.gaussian`. I think this corresponds to equation 2.9 in https://gaussianprocess.org/gpml/chapters/RW2.pdf. The calculation itself certainly seems to corresponds to equation 2.34 (though positive rather than negative).
